audio/render_thread.rs

```diff
@@ -160,10 +160,21 @@ impl AudioRenderThread {
         sample_rate: f32,
         graph: AudioGraph,
         options: AudioContextOptions,
+        init_sender: Sender<Result<(), AudioSinkError>>,
     ) {
-        let mut thread = Self::prepare_thread::<B>(sender.clone(), sample_rate, graph, options)
-            .expect("Could not start audio render thread");
-        thread.event_loop(event_queue)
+        let mut thread =
+            match Self::prepare_thread::<B>(sender.clone(), sample_rate, graph, options) {
+                Ok(thread) => {
+                    let _ = init_sender.send(Ok(()));
+                    thread
+                }
+                Err(e) => {
+                    let _ = init_sender.send(Err(e));
```

Comment:
@jdm Thanks for the guidance! I refactored the code to send an error when prepare_thread returns an error. The tests are no longer timing out locally. Can you please review it again? Thanks!

Reply:
Hmm, it's not timing out locally, but it's still timing out on CI:

> Test results for linux-wpt-layout-2020 from try job ([#10431866463](https://github.com/Taym95/servo/actions/runs/10431866463)):
> 
> Flaky unexpected result (17)
> Stable unexpected results that are known to be intermittent (13)
> Stable unexpected results (1)
> * TIMEOUT [expected OK] `/fetch/metadata/audio-worklet.https.html`

Reply:
never-mind I was using old commit on CI!